### ISSUE TITLE
nxos_command paramiko.ssh_exception.BadHostKeyException bad key exception failure

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29486]
From @ebmarquez on 2016-11-16T07:12:36Z

##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

  nxos_command
##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file = /home/emarq/Solutions.Network.Automation/MAS/Ansible/cisco/nexus/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION

```
[defaults]
hostfile=localstage
#hostfile=mas-b43
ansible_ssh_user=admin
ansible_ssh_private_key_file=/home/emarq/.ssh/masd-rsa
host_key_checking=False
```
##### SUMMARY

An updated switch configuration is sent to the switch.  The switch get's reloaded.  Post reload the ssh server key has changed on the switch.  The public key that is stored in known_hosts is no longer valid.  paramiko throws an error and fails.  in the Ansible config file, the host_key_checking=False is set.
##### STEPS TO REPRODUCE

```
---
- name:  Delete remote config
  nxos_command:
    provider: "{{ cli }}"
    host: "{{ ansible_host }}"
    commands:
      - "delete bootflash:{{ inventory_hostname }}.conf "

```

##### EXPECTED RESULTS

ignore ssh key change and establish a connection.  
##### ACTUAL RESULTS

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_sespbZ/ansible_module_nxos_command.py", line 257, in <module>
    main()
  File "/tmp/ansible_sespbZ/ansible_module_nxos_command.py", line 193, in main
    supports_check_mode=True)
  File "/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/network.py", line 112, in __init__
  File "/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/network.py", line 148, in connect
  File "/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/nxos.py", line 266, in connect
  File "/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/shell.py", line 226, in connect
  File "/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/shell.py", line 96, in open
  File "/usr/lib/python2.7/dist-packages/paramiko/client.py", line 353, in connect
    raise BadHostKeyException(hostname, server_key, our_server_key)
paramiko.ssh_exception.BadHostKeyException: ('10.10.230.12', <paramiko.rsakey.RSAKey object at 0x7f70d1c3f090>, <paramiko.rsakey.RSAKey object at 0x7f70d1c32b90>)

fatal: [rr1-n35-r10-x32sp-2a]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_name": "nxos_command"
    },
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_sespbZ/ansible_module_nxos_command.py\", line 257, in <module>\n    main()\n  File \"/tmp/ansible_sespbZ/ansible_module_nxos_command.py\", line 193, in main\n    supports_check_mode=True)\n  File \"/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/network.py\", line 112, in __init__\n  File \"/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/network.py\", line 148, in connect\n  File \"/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/nxos.py\", line 266, in connect\n  File \"/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/shell.py\", line 226, in connect\n  File \"/tmp/ansible_sespbZ/ansible_modlib.zip/ansible/module_utils/shell.py\", line 96, in open\n  File \"/usr/lib/python2.7/dist-packages/paramiko/client.py\", line 353, in connect\n    raise BadHostKeyException(hostname, server_key, our_server_key)\nparamiko.ssh_exception.BadHostKeyException: ('10.10.230.12', <paramiko.rsakey.RSAKey object at 0x7f70d1c3f090>, <paramiko.rsakey.RSAKey object at 0x7f70d1c32b90>)\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE"
}
```
ssh response to the key
```
emarq@rr1masdansible:~/Solutions.Network.Automation/MAS/Ansible/cisco/nexus$ ssh admin@10.10.230.13 -vvvvv
OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g  1 Mar 2016
debug1: Reading configuration data /home/emarq/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug2: resolving "10.10.230.13" port 22
debug2: ssh_connect_direct: needpriv 0
debug1: Connecting to 10.10.230.13 [10.10.230.13] port 22.
debug1: Connection established.
debug1: identity file /home/emarq/.ssh/masd-rsa type 1
debug1: key_load_public: No such file or directory
debug1: identity file /home/emarq/.ssh/masd-rsa-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_7.2p2 Ubuntu-4ubuntu2.1
debug1: ssh_exchange_identification: dcos_sshd run in non-FIPS mode


debug1: Remote protocol version 2.0, remote software version OpenSSH_6.2 FIPS
debug1: match: OpenSSH_6.2 FIPS pat OpenSSH* compat 0x04000000
debug2: fd 3 setting O_NONBLOCK
debug1: Authenticating to 10.10.230.13:22 as 'admin'
debug3: hostkeys_foreach: reading file "/home/emarq/.ssh/known_hosts"
debug3: record_hostkey: found key type RSA in file /home/emarq/.ssh/known_hosts:12
debug3: load_hostkeys: loaded 1 keys from 10.10.230.13
debug3: order_hostkeyalgs: prefer hostkeyalgs: ssh-rsa-cert-v01@openssh.com,rsa-sha2-512,rsa-sha2-256,ssh-rsa
debug3: send packet: type 20
debug1: SSH2_MSG_KEXINIT sent
debug3: receive packet: type 20
debug1: SSH2_MSG_KEXINIT received
debug2: local client KEXINIT proposal
debug2: KEX algorithms: curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1,ext-info-c
debug2: host key algorithms: ssh-rsa-cert-v01@openssh.com,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519
debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc,3des-cbc
debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc,3des-cbc
debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
debug2: compression ctos: none,zlib@openssh.com,zlib
debug2: compression stoc: none,zlib@openssh.com,zlib
debug2: languages ctos:
debug2: languages stoc:
debug2: first_kex_follows 0
debug2: reserved 0
debug2: peer server KEXINIT proposal
debug2: KEX algorithms: diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1
debug2: host key algorithms: ssh-rsa
debug2: ciphers ctos: aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc,rijndael-cbc@lysator.liu.se,aes128-ctr,aes192-ctr,aes256-ctr
debug2: ciphers stoc: aes128-cbc,3des-cbc,aes192-cbc,aes256-cbc,rijndael-cbc@lysator.liu.se,aes128-ctr,aes192-ctr,aes256-ctr
debug2: MACs ctos: hmac-sha1
debug2: MACs stoc: hmac-sha1
debug2: compression ctos: none,zlib@openssh.com
debug2: compression stoc: none,zlib@openssh.com
debug2: languages ctos:
debug2: languages stoc:
debug2: first_kex_follows 0
debug2: reserved 0
debug1: kex: algorithm: diffie-hellman-group-exchange-sha256
debug1: kex: host key algorithm: ssh-rsa
debug1: kex: server->client cipher: aes128-ctr MAC: hmac-sha1 compression: none
debug1: kex: client->server cipher: aes128-ctr MAC: hmac-sha1 compression: none
debug3: send packet: type 34
debug1: SSH2_MSG_KEX_DH_GEX_REQUEST(2048<7680<8192) sent
debug3: receive packet: type 31
debug1: got SSH2_MSG_KEX_DH_GEX_GROUP
debug2: bits set: 1017/2048
debug3: send packet: type 32
debug1: SSH2_MSG_KEX_DH_GEX_INIT sent
debug3: receive packet: type 33
debug1: got SSH2_MSG_KEX_DH_GEX_REPLY
debug1: Server host key: ssh-rsa SHA256:KwU71WtBGcBXk4XddRCstz7qTsTmPFxOPISEmDq1mN8
debug3: hostkeys_foreach: reading file "/home/emarq/.ssh/known_hosts"
debug3: record_hostkey: found key type RSA in file /home/emarq/.ssh/known_hosts:12
debug3: load_hostkeys: loaded 1 keys from 10.10.230.13
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the RSA key sent by the remote host is
SHA256:KwU71WtBGcBXk4XddRCstz7qTsTmPFxOPISEmDq1mN8.
Please contact your system administrator.
Add correct host key in /home/emarq/.ssh/known_hosts to get rid of this message.
Offending RSA key in /home/emarq/.ssh/known_hosts:12
  remove with:
  ssh-keygen -f "/home/emarq/.ssh/known_hosts" -R 10.10.230.13
RSA host key for 10.10.230.13 has changed and you have requested strict checking.
Host key verification failed.
```
Copied from original issue: ansible/ansible-modules-core#5628

**Comment by ansibot** [comment]
From @ansibot on 2016-11-16T07:12:36Z



@jedelman8, @GGabriele, @privateip, @gundalow, @qalthos, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mikewiebe @rahushen @rcarrillocruz @trishnaguha
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @kedarX
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by trishnaguha** [comment]
@ebmarquez lots of changes have been introduced in 2.4.0 release. Please give the module a try on the newer version and let us know if the issue still exists. 
needs_info

**Comment by ansibot** [comment]
@ansibot This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ebmarquez This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ebmarquez You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
